### PR TITLE
Add a non-distribution dogstatsd experiment

### PR DESCRIPTION
**Pull request by GeorgeHahn**
### What does this PR do?

Add a regression detector experiment for DogStatsD without any distribution metrics.

### Motivation

This will help monitor DogStatsD throughput with and without distributions in use.

### CODE REVIEW

File: test/regression/cases/uds_dogstatsd_to_api_nodist/lading/lading.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  - unix_datagram:
+      seed: [2, 3, 5, 7, 11, 13, 17, 19, 23, 29, 31, 37, 41, 43, 47, 53,
+             59, 61, 67, 71, 73, 79, 83, 89, 97, 101, 103, 107, 109, 113, 127, 131]
+      path: "/tmp/dsd.socket"
+      variant:
+        dogstatsd:
+          context_range: 1000-8000
+          name_length_range: 32-128
+          tag_key_length_range: 1-128
+          tag_value_length_range: 1:128
```
Comment (GeorgeHahn) [inline test/regression/cases/uds_dogstatsd_to_api_nodist/lading/lading.yaml:11]: This format is either wrong or surprising

Comment (blt) [reply test/regression/cases/uds_dogstatsd_to_api_nodist/lading/lading.yaml:11]: Hmm I also find that surprising.

**Comment by blt** [comment]
@GeorgeHahn Quite the difference, [ref](https://app.datadoghq.com/dashboard/t3q-y5d-i7u/smp-regression-detector-capture-data?tpl_var_experiment%5B0%5D=uds_dogstatsd_to_api&tpl_var_experiment%5B1%5D=uds_dogstatsd_to_api_nodist&tpl_var_run-id%5B0%5D=53db4deb-624e-48e3-88e0-0396b072b5d5&from_ts=1693589764219&to_ts=1693590261689&live=false).

**Review by blt — APPROVED** [review]

File: test/regression/cases/uds_dogstatsd_to_api_nodist/lading/lading.yaml (hunk 1 above)
Comment (GeorgeHahn) [reply test/regression/cases/uds_dogstatsd_to_api_nodist/lading/lading.yaml:11]: Ah, these keys are incorrect: https://github.com/DataDog/lading/blob/9d604dbcf96a0d6514930ac35f18feeb4311a278/lading_payload/src/dogstatsd.rs#L123

**Pull request merged**